query/src/lib/query-client-options.ts

```diff
@@ -11,11 +15,26 @@ export const QUERY_CLIENT_OPTIONS = new InjectionToken<QueryClientConfig>(
   },
 );
 
-export function provideQueryClientOptions(options: QueryClientConfig) {
+export type QueryClientConfigFn = () => QueryClientConfig;
+
+export function provideQueryClientOptions(
+  options: QueryClientConfig,
+): EnvironmentProviders;
+export function provideQueryClientOptions(
+  options: QueryClientConfigFn,
+): EnvironmentProviders;
+export function provideQueryClientOptions(
+  options: QueryClientConfig | QueryClientConfigFn,
+) {
   return makeEnvironmentProviders([
-    {
-      provide: QUERY_CLIENT_OPTIONS,
-      useValue: options,
-    },
+    typeof options === 'function'
+      ? {
+          provide: QUERY_CLIENT_OPTIONS,
+          useFactory: options,
```

Comment:
Make the bundle smaller :)

```suggestion
         [typeof options === 'function'  ? 'useFactory' : 'useValue']: options,
```

Reply:
It's me or is not working?
`Type '{ [x: string]: QueryClientConfig | InjectionToken<QueryClientConfig> | QueryClientConfigFn; provide: InjectionToken<...>; }' is not assignable to type 'Provider | EnvironmentProviders'.
  Type '{ [x: string]: QueryClientConfig | InjectionToken<QueryClientConfig> | QueryClientConfigFn; provide: InjectionToken<...>; }' is missing the following properties from type 'Type<any>': apply, call, bind, prototype, and 5 more.`

The best i'm able to do is this: `useFactory: typeof options === 'function' ? options : () => options,` but it's not the same, we are forcing useFactory instead of useValue to everyone.

Reply:
I don't mind to use `as any` even if it's solved the issue